### PR TITLE
feat(core/button) loading state

### PR DESCRIPTION
**Pull request by goncalosard**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Unit tests (`yarn test`) were run locally and passed
- [ ] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [ ] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bug fix
- [x] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

[IX-250]

## What is the new behavior?

- Added loading button example to Button documentation

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: packages/angular-test-app/src/preview-examples/button-loading.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-example',
+  template: `
+    <ix-button id="btn" (click)="loading()" class="m-1" variant="Secondary">
```
Comment (nuke-ellington) [inline packages/angular-test-app/src/preview-examples/button-loading.ts:15]: There should be three variations of the loading button scenario:

1. Button with text (already done)
2. Button with icon + text
3. Icon button

File: packages/angular-test-app/src/preview-examples/button-loading.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      Button
+    </ix-button>
+  `,
+})
+export default class ButtonLoading {
+  loading() {
+    var btn = document.getElementById('btn');
+    btn!.innerHTML =
+      'Loading... <ix-spinner variant="secondary" size="small"></ix-spinner>';
```
Comment (nuke-ellington) [inline packages/angular-test-app/src/preview-examples/button-loading.ts:24]: The order should be the other way around:

<ix-spinnter /> Loading

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

**Comment by goncalosard** [comment]
@nuke-ellington @danielleroux New loading button was added with the new props. Documentation examples were also added.
No VRT was added because of moving spinner, making it always fail.

**Comment by danielleroux** [comment]
Changed the PR to a breaking change.

I have replaced the spinner animation with a new one which fits better into the library. Final adaptions regarding the animation will be done maybe later on.

Removed the type from the interface of the spinner

Add a breaking change section to the document.

**Pull request merged**